package lapis;

message GroupConfig {
  message Group{
    optional int32 leader=1;
    repeated int32 member=2;
  }
  repeated Group group=1;
}

message SystemProto{
  //start rank for distributed memory table
  optional int32 table_server_start = 1 ;
  //end rank for distributed memory table
  optional int32 table_server_end = 2 ;
  // one node mode, no coordinator/memory servers
  optional bool standalone = 3 [default =true];
  // synschronous update if true
  optional bool synchronous = 4 [default = true];
}
